dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -113,10 +114,16 @@ private void logRetry(Throwable throwable) {
 
         private ListenableFuture<Response> wrap(ListenableFuture<Response> input) {
             return Futures.catchingAsync(
-                    Futures.transformAsync(input, this::success, MoreExecutors.directExecutor()),
+                    Futures.transformAsync(wrapWithTracing(input), this::success, MoreExecutors.directExecutor()),
                     Throwable.class,
                     this::failure,
                     MoreExecutors.directExecutor());
         }
+
+        private ListenableFuture<Response> wrapWithTracing(ListenableFuture<Response> input) {
+            DetachedSpan detachedSpan = DetachedSpan.start(String.format("request-attempt-%d", failures));
```

Comment:
```suggestion
            DetachedSpan detachedSpan = DetachedSpan.start("request-attempt-" + failures);
```

String.format is slow. compiling for jdk11 presizes string concatenation (even better than java8 automatic string builders)
https://openjdk.java.net/jeps/280